File: Chap1-Events/1-4EventPropagationDemo/childbutton.cpp
#include "childbutton.h"
#include <QDebug>

ChildButton::ChildButton(QWidget *parent) : ParentButton(parent)
{

}

void ChildButton::mousePressEvent(QMouseEvent *event)
{
    qDebug() << "ChildButton mousePressEvent called";
    ParentButton::mousePressEvent(event);
}

File: Chap1-Events/1-4EventPropagationDemo/childbutton.py
from parentbutton import ParentButton
from PySide6.QtGui import QMouseEvent

class ChildButton(ParentButton):
    def __init__(self, parent=None):
        super().__init__(parent)
    
    def mousePressEvent(self, event: QMouseEvent):
        print("ChildButton mousePressEvent called")
        super().mousePressEvent(event)